new makefile for SDDC_FX3 based on FX3 SDK examples from Cypress version 1.3.4

fventuri opened the pull request:
Howard, Oscar,
I just created a new makefile for the SDDC_FX3 firmware based on the EZ-USB FX3 SDK examples from Cypress version 1.3.4.

I ran it and it built the SDDC_FX3.img image file without errors here on Linux; please give it a try and let me know what you think.

Franco

howard0su commented:
Can you update README as well? I will try to download latest FX3 SDK and try this.

howard0su commented on SDDC_FX3/makefile:
```diff
@@ -47,25 +55,32 @@ EXES = $(MODULE).$(EXEEXT)
 $(MODULE).$(EXEEXT): $(A_OBJECT) $(C_OBJECT)
 	$(LINK)
 
+cyfxtx.c:
```
can we avoid copy here? Mixing the code and some .c file copied in is confusing. I can easy make mistake to add them to git as well. The current makefile will compile the file at its original place.

howard0su replied:
at least add them to .gitignore

fventuri replied:
The README now contains instructions on how to build the firmware (if you think they are too confusing, I can split the Windows installation of the SDK and the Linux installation of the SDK into two different subsections).

As per the two files ('cyfxtx.c' and 'cyfx_gcc_startup.S') that are being copied, I ran a 'diff' between the ones in the repo and the ones in Cypress EZ-USB FX3 SDK version 1.3.4, and the second one is nearly identical (except for the Copyright year), while the first one 'cyfxtx.c' has many significant changes, so perhaps it should be updated too.

howard0su commented:
Is it possible to auto compile the sdk and elf2img? This will reduce the amount of document to read. I am lazy.

fventuri commented:
Howard,
I added back in the new makefile the steps to recompile (and clean) 'elf2img' in the SDDC_FX3 directory every time, like the old makefile used to do.

I also updated both Cypress SDK files 'cyfxtx.c' and 'cyfx_gcc_startup.S' to the ones that come with the latest version of the EZ-USB FX3 SDK, as I mentioned last night.

I was able to compile everything here without errors; hopefully it works on Windows too.

ik1xpv commented:
Franco, my knowledge of makefile is quite low.  I'm looking at Howard for
reviewing.


On Mon, Nov 30, 2020 at 2:02 PM Franco Venturi <notifications@github.com>
wrote:

> Howard,
> I added back in the new makefile the steps to recompile (and clean)
> 'elf2img' in the SDDC_FX3 directory every time, like the old makefile used
> to do.
>
> I also updated both Cypress SDK files 'cyfxtx.c' and 'cyfx_gcc_startup.S'
> to the ones that come with the latest version of the EZ-USB FX3 SDK, as I
> mentioned last night.
>
> I was able to compile everything here without errors; hopefully it works
> on Windows too.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ik1xpv/ExtIO_sddc/pull/41#issuecomment-735772350>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACLNBGEBDIE4JUZAHFECN5TSSOJXZANCNFSM4UGWQCHQ>
> .
>

fventuri commented:
Thanks Howard.
Oscar, unless you have any objections, I can merge this change tomorrow morning my time (i.e. at about 2pm italian time).

Franco

ik1xpv commented:
Franco,
I leave you the pleasure to merge ! thanks

fventuri commented:
Done.
I just ran it here and I was able to compile the firmware without errors; I also edited the instructions on the NextGenSDRs group, and now they are much simpler.

Franco

howard0su commented:
After the switching to the new SDK, my RX888 was detected as BBRF103. We
may need to dig it out.

On Wed, Dec 2, 2020 at 8:48 PM Franco Venturi <notifications@github.com>
wrote:

> Merged #41 <https://github.com/ik1xpv/ExtIO_sddc/pull/41> into master.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ik1xpv/ExtIO_sddc/pull/41#event-4062170271>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF3GRFCLXPPW4POVNG3D7DSSYZSJANCNFSM4UGWQCHQ>
> .
>


-- 
-Howard

fventuri commented:
Howard,
sorry to hear that.

Since I have to start working now, feel free to revert  that commit, and we can look at it again in the next few days.

Franco

howard0su commented:
Don’t worry I will look into the issue. Revert is last choice.

Franco Venturi <notifications@github.com>于2020年12月2日 周三下午9:37写道：

> Howard,
> sorry to hear that.
>
> Since I have to start working now, feel free to revert that commit, and we
> can look at it again in the next few days.
>
> Franco
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/ik1xpv/ExtIO_sddc/pull/41#issuecomment-737234528>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF3GRDYDWJGSK4LCOBL3ETSSY7ILANCNFSM4UGWQCHQ>
> .
>
-- 
-Howard

howard0su commented:
diff --git a/SDDC_FX3/RunApplication.c b/SDDC_FX3/RunApplication.c
index a9b3034..7a1ab73 100644
--- a/SDDC_FX3/RunApplication.c
+++ b/SDDC_FX3/RunApplication.c
@@ -183,12 +183,12 @@ void ApplicationThread ( uint32_t input)
                if (I2cTransfer(0, R820T_I2C_ADDR, 1, &identity, true) ==
CY_U3P_SUCCESS)
                {
                        // check if BBRF103 or RX888 (RX666 ?)
-                       if(GPIOtestInputPulldown(LED_KIT)) {
+                       if(GPIOtestInputPulldown(LED_KIT) == CyTrue) {
                                HWconfig = BBRF103;

This fixes the problem. but I don't understand why. I think they are same
as CyTrue is 1.

On Wed, Dec 2, 2020 at 9:44 PM Howard Su <howard0su@gmail.com> wrote:

> Don’t worry I will look into the issue. Revert is last choice.
>
> Franco Venturi <notifications@github.com>于2020年12月2日 周三下午9:37写道：
>
>> Howard,
>> sorry to hear that.
>>
>> Since I have to start working now, feel free to revert that commit, and
>> we can look at it again in the next few days.
>>
>> Franco
>>
>> —
>> You are receiving this because your review was requested.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ik1xpv/ExtIO_sddc/pull/41#issuecomment-737234528>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAF3GRDYDWJGSK4LCOBL3ETSSY7ILANCNFSM4UGWQCHQ>
>> .
>>
> --
> -Howard
>


-- 
-Howard
